Models/Consultants/ConsultantsRepository.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace cautious_waddle.Models
+{
+    public class ConsultantsRepository : IConsultantsRepository
+    {
+        ConsultantsDbContext _context;
+
+        public ConsultantsRepository(ConsultantsDbContext context)
+        {
+            _context = context;
+        }
+
+        public string GetUserId(int id)
+        {
+            return _context.Consultants.SingleOrDefault(c => c.ConsultantId == id).UserId;
+        }
+
+        public Consultant GetConsultantById(int id)
+        {
+            return _context.Consultants.SingleOrDefault(c => c.ConsultantId == id);
+        }
+
+        public IEnumerable<Consultant> GetConsultants()
+        {
+            IEnumerable<Consultant> consultants = _context.Consultants;
+            return consultants;
+        }
+
+        public void AddConsultant(Consultant consultant)
+        {
+            _context.Consultants.Add(consultant);
+            _context.SaveChanges();
+        }
+
+        public void EditConsultant(Consultant consultant)
+        {
+            Consultant oldConsultant = GetConsultantById(consultant.ConsultantId.Value);
+            _context.Consultants.Attach(oldConsultant);
+
+            oldConsultant.LastUpdate     = DateTime.Now;
```

Comment:
@millar-98  I think I have automapper installed . You could potentially use it to map old objects to new. E.g Mapper.Map<Source, Destination>(source, destination);

Reply:
btw this isnt a blocking it from going live just something if youre interested